flowmachine/flowmachine/core/cache.py

```diff
@@ -88,14 +89,54 @@ def column_names(self):
         )
 
 
+def run_ops_list_and_return_execution_time(
+    query_ddl_ops: List[str], connection: Engine
+) -> float:
+    """
+    Run a list of sql operations which may begin with explain analyze
+    and return the overall execution time of the first op if available.
+
+    Parameters
+    ----------
+    query_ddl_ops : list of str
+        List of SQL commands to execute
+    connection : Engine
+        SQLAlchemy engine to execute with
+
+    Returns
+    -------
+    float
+        The execution time of the query if available or 0
+    """
+    plan_time = 0
+    ddl_op_results = []
+    for ddl_op in query_ddl_ops:
+        try:
+            ddl_op_result = connection.execute(ddl_op)
+        except Exception as e:
+            logger.error(f"Error executing SQL: '{ddl_op}'. Error was {e}")
+            raise e
+        try:
+            ddl_op_results.append(ddl_op_result.fetchall())
+        except ResourceClosedError:
+            pass  # Nothing to do here
+        for ddl_op_result in ddl_op_results:
```

Comment:
I haven't looked at this in detail before, and I'm not sure I understand what it's doing. We're looping over all DDL ops and appending the results to `ddl_op_results`, but within the loop we're looping over `ddl_op_results` and adding the execution time s to the overall `plan_time`. So execution times of earlier ops will be counted multiple times, if there are multiple ops?

Reply:
Good catch, I've screwed up an indent while moving it out to a function I think.